apps/ccf-rui/src/app/modules/registration-modal/registration-content/registration-content.component.ts

```diff
@@ -57,57 +84,116 @@ export class RegistrationContentComponent {
   constructor(
     readonly page: PageState,
     readonly model: ModelState,
+    readonly registration: RegistrationState,
     public dialogRef: MatDialogRef<RegistrationContentComponent>,
     cdr: ChangeDetectorRef,
   ) {
```

Comment:
You need to clean up the subscriptions that are created in the constructor to prevent resource leaks. Use https://angular.dev/api/core/rxjs-interop/takeUntilDestroyed

```ts
page.user$.pipe(takeUntilDestroyed()).subscribe(...)
```